src/Io/RequestHeaderParser.php

```diff
@@ -24,20 +25,48 @@ class RequestHeaderParser extends EventEmitter
 {
     private $maxSize = 8192;
 
+    /**
+     * @var LoopInterface
+     */
+    private $loop;
+
     /** @var Clock */
     private $clock;
 
-    public function __construct(Clock $clock)
+    /**
+     * @var float
+     */
+    private $idleConnectionTimeout;
+
+    /**
+     * @param LoopInterface $loop
+     * @param float $idleConnectionTimeout
+     */
+    public function __construct(LoopInterface $loop, Clock $clock, $idleConnectionTimeout)
     {
+        $this->loop = $loop;
         $this->clock = $clock;
+        $this->idleConnectionTimeout = $idleConnectionTimeout;
     }
 
     public function handle(ConnectionInterface $conn)
     {
+        $loop = $this->loop;
+        $idleConnectionTimeout = $this->idleConnectionTimeout;
+        $timer = $loop->addTimer($idleConnectionTimeout, function () use ($conn) {
+            $conn->close();
```

Comment:
Should this trigger an `error` event similar to how an overflow would trigger?

Reply:
Done